app/jobs/scheduled/periodical_updates.rb

```diff
@@ -25,7 +25,11 @@ def execute(_ = nil)
 
       # Forces rebake of old posts where needed, as long as no system avatars need updating
       if !SiteSetting.automatically_download_gravatars ||
-           !UserAvatar.where("last_gravatar_download_attempt IS NULL").limit(1).first
+           !UserAvatar
+             .joins(user: :user_emails)
+             .where(user_emails: { primary: true })
+             .where(last_gravatar_download_attempt: nil)
```

Comment:
```suggestion
             .where(user_emails: { primary: true }, last_gravatar_download_attempt: nil)
```

Minor one but it is one less method call.